llm-examples/falcon/falcon-7b-instruct-inference.py

```diff
@@ -0,0 +1,349 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC # Falcon-7b-instruct Inference on Databricks
+# MAGIC Environment for this notebook:
+# MAGIC - Runtime: 13.1 GPU ML Runtime
+# MAGIC - Instance: `g5.4xlarge` on AWS
+# MAGIC
+# MAGIC GPU instances that have at least 16GB GPU memory would be enough for inference on single input (batch inference requires slightly more memory). On Azure, it is possible to use `Standard_NC6s_v3` or `Standard_NC4as_T4_v3`.
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Install required libraries
+
+# COMMAND ----------
+
+# MAGIC %pip install -q -U torch==2.0.1
+# MAGIC %pip install -q einops==0.6.1
+
+# COMMAND ----------
+
+dbutils.library.restartPython()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Inference
+# MAGIC The below snippets are adapted from [the model card of falcon-7b-instruct](https://huggingface.co/tiiuae/falcon-7b-instruct). The example in the model card should also work on Databricks with the same environment.
+
+# COMMAND ----------
+
+# Load model to text generation pipeline
+
+from transformers import AutoTokenizer, AutoModelForCausalLM
+import transformers
+import torch
+
+model = "tiiuae/falcon-7b-instruct"
+
+tokenizer = AutoTokenizer.from_pretrained(model)
+pipeline = transformers.pipeline(
+    "text-generation",
+    model=model,
+    tokenizer=tokenizer,
+    torch_dtype=torch.bfloat16,
+    trust_remote_code=True,
+    device_map="auto",
+    revision="9f16e66a0235c4ba24e321e3be86dd347a7911a0", # it is a suggested practice to pin the revision commit hash
+)
+
+# COMMAND ----------
+
+# Define prompt template, see why we need this format: http://fastml.com/how-to-train-your-own-chatgpt-alpaca-style-part-one/
+INSTRUCTION_KEY = "### Instruction:"
+RESPONSE_KEY = "### Response:"
+INTRO_BLURB = "Below is an instruction that describes a task. Write a response that appropriately completes the request."
+PROMPT_FOR_GENERATION_FORMAT = """{intro}
+{instruction_key}
+{instruction}
+{response_key}
+""".format(
+    intro=INTRO_BLURB,
+    instruction_key=INSTRUCTION_KEY,
+    instruction="{instruction}",
+    response_key=RESPONSE_KEY,
+)
+
+# COMMAND ----------
+
+# Define parameters to generate text
+def gen_text(prompt, use_template=False, **kwargs):
+    if use_template:
+      full_prompt = PROMPT_FOR_GENERATION_FORMAT.format(instruction=prompt)
+    else:
+      full_prompt = prompt
+    
+    # configure text generation kwargs
+    if "max_new_tokens" not in kwargs:
+        kwargs["max_new_tokens"] = 512
+    
+    kwargs.update({
+      'do_sample': True,
+      'use_cache': True,
+      'num_return_sequences': 1,
+      'pad_token_id': tokenizer.eos_token_id,
+      'eos_token_id': tokenizer.eos_token_id
+    })
```

Comment:
also can you put more explanation on the arguments. especially what's the effect of setting them so that user can tune those according to their needs when they do the DIY